Description: Fix Zabbix errors [Z3005] query failed: [2006] MySQL server has
gone away, using auto_reconnect feature from libmysqlclient
Author: Etienne CHAMPETIER
Origin: other, https://support.zabbix.com/browse/ZBX-6163
Reviewed-by: Jean Baptiste Favre <jean-baptiste.favre@blablacar.com>
Last-Update: 2013-11-21
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: zabbix/src/libs/zbxdb/db.c
===================================================================
--- zabbix.orig/src/libs/zbxdb/db.c	2016-01-23 13:09:06.402938250 +0100
+++ zabbix/src/libs/zbxdb/db.c	2016-01-23 13:09:06.398938230 +0100
@@ -316,6 +316,9 @@
 	{
 		zabbix_log(LOG_LEVEL_CRIT, "cannot allocate or initialize MYSQL database connection object");
 		exit(EXIT_FAILURE);
+	}else{
+		my_bool reconnect = 1;
+		mysql_options(conn, MYSQL_OPT_RECONNECT, &reconnect);
 	}
 
 	if (NULL == mysql_real_connect(conn, host, user, password, dbname, port, dbsocket, CLIENT_MULTI_STATEMENTS))
